### DISCUSSION TITLE
Quick Google search link

### DISCUSSION THREAD

**Discussion by luckman212** [SilentVoid13/Templater#142] [Templates Showcase]
Quickly insert a Google Search link as a markdown link.

```js
<%*
  if (tp.file.selection() === "" ) {
    new Notice('Please select some text first!');
  } else {
    %>[<% tp.file.selection() %>](https://google.com/search?q=<% encodeURIComponent(tp.file.selection()) %>)<%*
  }
%>
```

**Comment by welpdx** [comment]
```js
<%*
  if (tp.file.selection() === "" ) {
    new Notice('Please select some text first!');
  } else {
    %>[<% tp.file.selection() %>](https://scholar.google.com/scholar?q=<% encodeURIComponent(tp.file.selection()) %>)<%*
  }
%>
```

If you want google scholar search

**Comment by nyanmaid** [comment]
Why do we escape tp.file.selection() and encodeURIComponent? I think these are available in js without being escaped out from the script. 
Also, we can use tR to output texts, is that correct?

**Reply by AB1908** [reply]
You can and there's many ways of writing the text. I personally prefer a slightly different approach but if it works, it works.

**Reply by MarredCheese** [reply]
OP chose to keep switching in and out of literal mode and code mode to combine static and dynamic text.  That's why it's like that.  But I think it's unnecessarily confusing.  Here's something more straightforward:

```js
<%*
const s = tp.file.selection();
if (s === "")
  new Notice('Please select some text first!');
else
  tR += `[${s}](https://google.com/search?q=${encodeURIComponent(s)})`;
%>
```

**Reply by JustAnotherSapiens** [reply]
The Bing version:
```js
<%*
  const selection = tp.file.selection();
  if (selection === '') {
    new Notice('Please select some text first!');
  } else {
    const searchQuery = encodeURIComponent(selection);
    var bingSearch = `https://www.bing.com/search?q=${searchQuery}`;
    tR += `[${selection}](${bingSearch})`;
  }
%>
```

**Comment by Dercraker** [comment]
It's easy to update this script by replacing the notice with an entry.
This allows you to add search text if nothing is selected. 

```js
<%*
let searchText = tp.file.selection();
if (searchText === "")
  searchText = await tp.system.prompt("Please enter some search text...","",true);
const searchQuery = encodeURIComponent(searchText);
tR = `[${searchText}](https://google.com/search?q=${searchQuery})\n`;
%>
```

**Reply by iLiftALot** [reply]
``tR += `\n[${searchText}](https://google.com/search?q=${searchQuery})\n`;``

Ensure you add that `+=` in the last line to avoid losing page content
